src/chromo/models/pythia8.py

```diff
@@ -161,28 +173,61 @@ def _cross_section(self, kin=None):
             diffractive_axb=st.sigmaAXB,
         )
 
+        # TODO use pythia.info.sigmaGen(i) for cross-sections including nucleus in the
+        # initial collision system.
+        #   pythia.info.sigmaTot() are the estimated cross-section values for the
+        # equivalent pp collision for each sub-processes.
+        #   pythia.info.sigmaGen(i) are the cross-section values generated by the MC
+        # events for each subprocesses. Best is to grab these values at the end of the
+        # event generation when the full statistics is available.
+        # The subprocesses are defined with codes in Pythia:
+        #    i   subprocess
+
+    #   101  non-diffractive
+    #   102  A B -> A B elastic
+    #   103  A B -> X B single diffractive
+    #   104  A B -> A X single diffractive
+    #   105  A B -> X X double diffractive
+    #   106  A B -> A X B central diffractive
+
     def _set_kinematics(self, kin):
         config = self._config[:]
 
+        # Path prefix for some cache files that Pythia8 generates
+        # to speed up subsequent runs
+        cache_prefix = str(
+            _cache_base_dir()
+            / "Pythia8"
+            / f"cache_{int(kin.p1)}_{int(kin.p2)}_{kin.ecm / MeV:.0f}".replace("-", "m")
```

Comment:
what about the `+` or other characters?

One problem I can see is that a file is created for each center of mass energy. How is this going to behave if we use this in a cascade MC (what we’re doing elsewhere)? It means that an infinite number of files will be created.

I heard that Pythia8 has been made compatible somehow with CORSIKA 8? Then there must be some sort of way to get energy dependent tables. Caching projectile/target combinations is ok though since it must be limited.

Also…. why MeV?

Reply:
Thanks for the explanations. @gaudu so you mean that the `allowVariableEnergy` is broken but will be fixed in November? Would that be something we could use permanently in Chromo? We also are in November... so you'd prefer not to wait for this latest version?

For now, I think it is ok to generate infinite files. However, I would like to ask you to throw warnings if the number if files in the cache folder exceeds 10k or so. We already have at least one application that changes the "beam" configuration in every event and it would strain the file system immediately.

Reply:
Ok, we will implement a warning.

I would prefer to merge now and then switch to the other solution later. We have another patch of the cross-sections planned that is urgent fuer Chloe, which is based on this PR. It is a small patch to remove the cache file feature once a better solution is implemented in Pythia8 and works stable.

Reply:
Why can't she subclass for the time being? I have concerns about caching globally. If I want to preselect and xB or pT cut then the cache files are not valid. Or if a process is switched off, then the files are invalid. Pythia doesn't seem to check for the consistency.

From Pythia8 docs:
>  This can save a significant time, notably for short test runs. A warning is that all relevant settings must be the same in the two runs. It is up to the user to ensure that this is the case; it would be next-to-impossible to catch any possible conflict.

Reply:
@afedynitch This feature (in a bugged state or not) will be a later addition to our toolkit, or even better within the Pythia toolkit. I believe it should not be a reason to further delay this merge request. 

As Hans brought it up, there is further work to be done to bring to the heavy ions runs in chromo correct cross-sections from a Physics point-of-view. I am hoping to deal with this in another PR.

Reply:
Yes, but what is the reason to merge an unsafe PR? In `chromo`, all models run in default config, or as close as it can get. Any user can tinkle with whatever is exposed. But it is against the philosophy of such a generic code that targets many different user.

Also I don't understand the rationale behind the caching by default. If you need to create multiple pythia instances within the same interpreter session, why not using a tmp folder? Also, I don't get why this is such a critical feature. A user, producing 10-50k events for a single energy in pp can wait a few seconds to generate fresh tables.

For heavy ions, this might be different but since the heavy ion stuff will get bugfixed in the next version of Pythia, I strongly suggest to subclass for the time being and implement the latest version with some more generic treatment of the nuclear cases.

Reply:
Using the cache is not a dangerous feature and since it is a performance option, it is not going against your wish to run with defaults as much as possible. It is not an unsafe PR in my view.

Reply:
@HDembinski, if you change pT0, or the tune - this is something Pythia users do quite often - then the presence of an old cache file will generate undefined behavior because the cross section table doesn't match. The cache is transparent to the user (since its hidden in a "system" folder and there is no tracking of which option combination has been used to generate it. 

Artificial but not unrealistic example: I download chromo and run Pythia8 with my favorite pT0 setting of 2.567 GeV at a cms energy that's used in one of the tests. Then running the tests afterwards will likely fail because it will pick up a cache file generated with altered settings but apply it to a default run.

It is strange that you are in favor such an implementation.

I'm not against merging the more recent Pythia version with all the downloads and files etc, but I'm against merging the flawed cache implementation and the non-default default settings.